### ISSUE TITLE
navit 0.5.4 no longer reads custom navit.xml

### ISSUE THREAD

**Issue by mensch21** [navit-gps/navit#1030]
Hello,
the lastest play store navit version v0.5.4-fdroid.3-39-g8f45131 from 2020/07/13 no longer reads the custom navit.xml on /storage/emulated/0/navit/.

I went back to 0.5.1.
By the way... the former desktop icon was more known and more likely...

**Comment by gefin** [comment]
Did you grant permission to access storage ?
(settings->apps->navit...)

**Comment by jkoan** [comment]
> Did you grant permission to access storage ?
> (settings->apps->navit...)
> Navit dont ask for it like the older versions did.
> 
> Edit: just downloaded 5.4 from playstore, grant storage permisson in settings, set map storage to /sdcard/navit : custom navit.xml work well

Yes that's true. But i see that this isn't the solution for none technical people. We should probably ask when opening the directory select screen or something like this...

**Comment by mensch21** [comment]
Now I granted storage permission (after new download from playstore). I selected /mnt/sdcard/navit/NavitMaps/ for map storage but I get same results. This is "hidden functionaltity" and will cause problems by standard users.

navit.xml on /mnt/sdcard/navit/ has not been accessed and selected map storage /mnt/sdcard/navit/NavitMaps/ tells "Directory is empty. Using file browser I see Europe.bin (15GB loaded 2019/11/25 from http://maps3.navit-project.org/).

The path to my maps is in navit.xml (<maps type="binfile" data="/storage/emulated/0/navit/NavitMaps/*.bin"/>)
If I reinstall 0.5.1 everything works fine.

Then I selected /mnt/sdcard/navit/ for map storage (like jkoan wrote) and restarted navit again and now it works.

I think this is a bit tricky and very complicate for users who have not much experience in information technologies.
Ok, before retirement I was programmer, so more or less IT specialist. But I think most of the users are not.

And form now I have to select map storage directory where custom navit.xml is stored (/storage/emulated/0/navit/) but my maps I stored in /storage/emulated/0/navit/NavitMaps/ for more overview. This is confusing to set or select the map storage to let navit read custom navit.xml from this directory.

Why don't you let select navit.xml and map storage separately? But the former way was easier. When custom.xml was found in navit directory it has been taken. And all other stuff is defined in cutom navit.xml - should be...

In December 2017 you did not merge the .../nodpi/ folder (the icon-folder) into new navit.apk. I informed navit team about this situation and this problem was solved fast and easialy. But there were a lot of negative comments following in playstore. It would be better to keep the customers satisfied.

In my job as a programmer quite often I let my wife or some friend handle my software. This kept me closer to the costomers because they told me what is not clear for them. Even I learned that software developers fade out from user knowledge.

I will keep 0.5.4 and go on looking how it works. If I get problems I have the 0.5.1-apk on sd card.

Best regards
Achim (mensch21)

This is a screenshot from 0.5.4
![Screenshot_20200716-000141](https://user-images.githubusercontent.com/43501862/87602894-197c7f80-c6f8-11ea-8ca4-28bd0666c8c5.png)

**Comment by gefin** [comment]
I just use /sdcard/navit for map storage and navit.xml

The map location dialog shows "directory empty" but it work after restart navit.
